### ISSUE TITLE
Deploy Contract and Invoke Contract dialogs: counter-intuitive

### ISSUE THREAD

**Issue by mwherman2000** [CityOfZion/neo-gui-developer#39]
You can call it a usability or functionality or workflow issue but it appears that the contract deployment transaction doesn't happen (execute) until you click the **Test** button on the **Invoke Contract** dialog.  This is counter-intuitive and will lead to a lot of "user error". Checkout the function `btnTestScript_Click()` here:

[https://github.com/CityOfZion/neo-gui-developer/blob/d1cc87912c5899d83606ddcdf839a01a0b83d8b7/neo-gui/UI/InvokeContractDialog.cs#L254](https://github.com/CityOfZion/neo-gui-developer/blob/d1cc87912c5899d83606ddcdf839a01a0b83d8b7/neo-gui/UI/InvokeContractDialog.cs#L254)

People are going to expect the deployment should take place when you click the Deploy button in the Deploy Contract dialog.  This should be refactored.

**Comment by ghost** [comment]
Agreed, but this is the way it has been implemented in neo-project/neo-gui. For the sake of future updates I don't think it is a big enough problem to warrant changing the workflow.

Thoughts?
